src/coreclr/jit/morph.cpp

```diff
@@ -14349,6 +14432,13 @@ GenTree* Compiler::fgMorphMultiOp(GenTreeMultiOp* multiOp)
     }
 #endif // defined(FEATURE_HW_INTRINSICS) && defined(TARGET_XARCH)
 
+#ifdef FEATURE_HW_INTRINSICS
+    if (multiOp->OperIsHWIntrinsic() && !optValnumCSE_phase)
```

Comment:
why is ` !optValnumCSE_phase` check needed?

Reply:
It was added because I was deleting nodes; however, I'm not deleting the nodes anymore and instead using `ResetHWIntrinsic`. I do feel safer having that guard there in-case more optimizations get added in `fgOptimizeHWIntrinsic`.

Read more in the discussion: https://github.com/dotnet/runtime/pull/63821#discussion_r790040097